src/bundle.c

```diff
@@ -466,15 +462,32 @@ int install_bundles(struct list *bundles, int current_version, struct manifest *
 		}
 
 		ret = do_staging(file, mom);
-		if (ret == 0) {
-			rename_staged_file_to_final(file);
-		} else {
-			printf("Failed to stage file: %s (ret = %d). Aborting bundle-add\n",
-			       file->filename, ret);
+		if (ret) {
+			ret = verify_fix_path(file->filename, mom);
+		}
+		if (ret) {
+			ret = EBUNDLE_INSTALL;
 			goto out;
 		}
 	}
 
+	iter = list_head(to_install_files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+
+		if (file->is_deleted || file->do_not_update || ignore(file)) {
+			continue;
+		}
+
+		/* This was staged by verify_fix_path */
+		if (!file->staging) {
+			file = search_file_in_manifest(mom, file->filename);
+		}
```

Comment:
So this confused me at first too but it goes like this:
The to_install_files list is populated with a call to recurse_manifest that is separate from the the mom->files list (populated from recurse_manifest and then consolidated).

The rename_staged_file_to_final is going to look at the file structure and verify if it was staged or not (since this is bundle-add everything would be staged) but that structure is updated in do_staging which in the verify_fix_path case failed and so did not populate the file->staging string. verify_fix_path updates the file->staging string in the mom (using search_file_in_manifest) which is what I end up using here.

This could definitely have been further detailed but I'm actually in favor of changing how all this works to only have one copy of the file struct eventually.